contracts/strategies/ltip-simple/LTIPSimpleStrategy.sol

```diff
@@ -0,0 +1,708 @@
+    /// @param _data The data to be decoded
+    /// @param _sender The sender of the allocation
+    function _allocate(bytes memory _data, address _sender)
+        internal
+        virtual
+        override
+        nonReentrant
+        onlyActiveAllocation
+    {
+        // Decode the '_data'
+        address recipientId = abi.decode(_data, (address));
+        Recipient storage recipient = _recipients[recipientId];
+
+        if (recipient.recipientStatus != Status.Accepted) revert RECIPIENT_NOT_ACCEPTED();
+
+        // Check if the allocator has already casted the vote
+        address voteCastedTo = votedFor[_sender];
+        if (voteCastedTo != address(0)) {
+            // remove the old vote to allow recasting of vote
+            votes[voteCastedTo] -= 1;
+        }
+
+        // Increment the votes for the recipient
+        votes[recipientId] += 1;
+        // Update the votedFor mapping
+        votedFor[_sender] = recipientId;
+
+        // Emit the event
+        emit Voted(recipientId, _sender);
+
+        if (votes[recipientId] >= votingThreshold) {
+            emit Allocated(recipientId, recipient.allocationAmount, allo.getPool(poolId).token, address(0));
+        }
+    }
+
+    /// @notice Create a vesting plan for the recipient
+    /// @param recipientId ID of the recipient
+    /// @param recipientAddress Address of the recipient
+    /// @param _token The token to be vested
+    /// @param _amount The amount to be vested
+    function _vestAmount(address recipientId, address recipientAddress, address _token, uint256 _amount)
+        internal
+        virtual
+    {
+        TokenTimelock vestingContract =
+            new TokenTimelock(IERC20(_token), recipientAddress, block.timestamp + vestingPeriod);
+
+        _transferAmount(_token, address(vestingContract), _amount);
+
+        _vestingPlans[recipientId] = VestingPlan(address(vestingContract), 0);
+
+        emit VestingPlanCreated(recipientId, address(vestingContract), 0);
+    }
+
+    /// @notice Distribute the upcoming milestone to acceptedRecipientId.
+    /// @dev As allocation determines the acceptance, anybody should be able to distribute.
+    /// @param _sender The sender of the distribution.
+    function _distribute(address[] memory __recipientIds, bytes memory, address _sender) internal virtual override {
+        IAllo.Pool memory pool = allo.getPool(poolId);
+
+        uint256 recipientLength = __recipientIds.length;
+
+        for (uint256 i; i < recipientLength;) {
+            address recipientId = __recipientIds[i];
+
+            Recipient memory recipient = _recipients[recipientId];
+
+            // Check if the recipient is accepten
+            if (recipient.recipientStatus != Status.Accepted) revert RECIPIENT_NOT_ACCEPTED();
+            if (votes[recipientId] < votingThreshold) revert INSUFFICIENT_VOTES();
+
+            if (_vestingPlans[recipientId].vestingContract != address(0)) revert ALREADY_ALLOCATED();
```

Comment:
The error message is not very accurate, this checks if the recipient has a vesting. A better error name could be `ALREADY_VESTED` or `VESTING_CONTRACT_ALREADY_SET` ?

Reply:
ALREADY_VESTED